src/chains/algorand_1/templates/chainActions/chainSpecific/asset_config.ts

```diff
@@ -0,0 +1,53 @@
+import * as algosdk from 'algosdk'
+import {
+  AlgorandDecomposeReturn,
+  AlgorandChainActionType,
+  AlgorandSuggestedParams,
+  AlgorandTransactionTypeCode,
+  AlgorandTxAction,
+  AlgorandTxActionRaw,
+  AlgorandActionAssetConfigParams,
+} from '../../../models'
+import { AlgorandActionHelper } from '../../../algoAction'
+import { isNullOrEmpty } from '../../../../../helpers'
+
+/**
+ * Composes asset config action */
+export const composeAction = (args: AlgorandActionAssetConfigParams, suggestedParams: AlgorandSuggestedParams) => {
+  const argsEncodedForSdk = new AlgorandActionHelper(args as AlgorandTxAction).actionEncodedForSdk
+  const { from, note, assetIndex, assetManager, assetReserve, assetFreeze, assetClawback } = argsEncodedForSdk
+  const composedAction = algosdk.makeAssetConfigTxnWithSuggestedParams(
+    from,
+    note,
+    assetIndex,
+    assetManager,
+    assetReserve,
+    assetFreeze,
+    assetClawback,
+    suggestedParams,
+    args.strictEmptyAddressChecking,
+  )
+  return { ...composedAction }
+}
+
+export const decomposeAction = (action: AlgorandTxAction | AlgorandTxActionRaw): AlgorandDecomposeReturn => {
+  const actionHelper = new AlgorandActionHelper(action)
+  const actionParams = actionHelper.paramsOnly
+  // Identify chainActionType using type
+  if (
+    actionParams?.type === AlgorandTransactionTypeCode.AssetConfig &&
+    (!isNullOrEmpty(actionParams?.assetManager) ||
```

Comment:
This logic clashes with the logic in asset_destroy. If all these fields are provided in asset_destroy, then this decompose would say that that action is also AssetConfig because of the OR condition here. 
Look for logic that guarantees single action type for all decompose functions

Reply:
This one checks if any of the asset* property exists. The purpose of the assetConfig is to edit at least one of them. So I check if at least one of them is there. AssetDestroy check if none of them exists. (all nullOrEmpty)